4.3.js

```diff
@@ -0,0 +1,38 @@
+function arrayToList(array) {
+  var list = null;
+  for (var i = array.length - 1; i >= 0; i--) {
+    list = {value: array[i], rest: list};
+  }
+  return list;
+}
+
+function listToArray(list) {
+  var array = [];
+  for (var node = list; node; node = node.rest) {
+    array.push(node.value);
+  }
+  return array;
+}
+
+function prepend(element, list) {
+  result =  {value: element, rest: list};

```

Comment:
duzy blad - zapomnialas o slowie kluczowym "var". przez co result stal sie wlasnie zmienna globalna.
http://stackoverflow.com/questions/1470488/what-is-the-purpose-of-the-var-keyword-and-when-to-use-it-or-omit-it